recipes/lcms/all/conanfile.py

```diff
@@ -69,10 +69,16 @@ def _build_visual_studio(self):
         if tools.Version(self.version) <= "2.11":
             vc_sln_subdir = "VC2013"
         else:
-            vc_sln_subdir = "VC2015"
+            if Version(self.settings.compiler.version) == "14":
+                vc_sln_subdir = "VC2015"
+            elif Version(self.settings.compiler.version) == "15":
+                vc_sln_subdir = "VC2017"
+            else:
+                vc_sln_subdir = "VC2019"
+
         with tools.chdir(os.path.join(self._source_subfolder, "Projects", vc_sln_subdir )):
             target = "lcms2_DLL" if self.options.shared else "lcms2_static"
-            upgrade_project = Version(self.settings.compiler.version) > "12"
+            upgrade_project = Version(self.settings.compiler.version) > "16"
```

Comment:
think you need to keep the upgrade.
I had a similar problem with another library (I don't remember which), which I could work around by removing some lines out of the vcx project.
Perhaps [this line](https://github.com/mm2/Little-CMS/blob/c65934f44aaf6cfc1a37f30f379bb76d5c891ec8/Projects/VC2019/lcms2_static/lcms2_static.vcxproj#L25), but I don't guarantee anything.

First, let's try to upgrade the project, before patching the build script.

Reply:
Thanks for the recommendation - I'm adding the upgrade back in!